src/NUnitFramework/framework/Internal/Tests/TestMethod.cs

```diff
@@ -76,17 +78,19 @@ internal object ExpectedResult
         {
             get { return parms != null ? parms.ExpectedResult : null; }
         }
-
-        internal object[] Arguments
-        {
-            get { return parms != null ? parms.Arguments : null; }
-        }
-
         #endregion
 
         #region Test Overrides
 
         /// <summary>
+        /// The arguments to use in executing the test method.
+        /// </summary>
+        public override object[] Arguments
+        {
+            get { return parms != null ? parms.Arguments : NoArguments; }
```

Comment:
Is this change from returning `null` to `object[0]` definitely ok? This property is eventually used to invoke the method via `MethodBase.Invoke()` - which [states null should be used when there are no parameters](https://msdn.microsoft.com/en-us/library/a89hcwhh(v=vs.110).aspx). 

If all the tests pass, I assume it works, but I'm a little wary that I might be missing something!